src/ser.rs

```diff
@@ -190,17 +197,17 @@ where
     }
 }
 
-impl<'a, W> ser::Serializer for Serializer<'a, W>
+impl<'r, W> ser::Serializer for Serializer<&'r mut W>
```

Comment:
why `'r` ?

Reply:
Fixed in [dc996e2](https://github.com/zefchain/bcs/pull/9/commits/dc996e2618b25d7c97ce5575e35dc21b1cff4e4a).